src/anemoi/training/losses/mse.py

```diff
@@ -10,82 +10,72 @@
 from __future__ import annotations
 
 import logging
+from functools import cached_property
 
 import torch
-from torch import nn
+
+from anemoi.training.losses.weightedloss import WeightedLoss
 
 LOGGER = logging.getLogger(__name__)
 
 
-class WeightedMSELoss(nn.Module):
+class WeightedMSELoss(WeightedLoss):
     """Latitude-weighted MSE loss."""
 
     def __init__(
         self,
         node_weights: torch.Tensor,
-        data_variances: torch.Tensor | None = None,
-        ignore_nans: bool | None = False,
+        feature_weights: torch.Tensor | None = None,
+        ignore_nans: bool = False,
+        **kwargs,
     ) -> None:
-        """Latitude- and (inverse-)variance-weighted MSE Loss.
+        """Latitude- and feature weighted MSE Loss.
 
         Parameters
         ----------
         node_weights : torch.Tensor of shape (N, )
             Weight of each node in the loss function
-        data_variances : Optional[torch.Tensor], optional
-            precomputed, per-variable stepwise variance estimate, by default None
+        feature_weights : Optional[torch.Tensor], optional
+            precomputed, per-variable weights, by default None
         ignore_nans : bool, optional
             Allow nans in the loss and apply methods ignoring nans for measuring the loss, by default False
 
         """
-        super().__init__()
-
-        self.avg_function = torch.nanmean if ignore_nans else torch.mean
-        self.sum_function = torch.nansum if ignore_nans else torch.sum
-
-        self.register_buffer("weights", node_weights, persistent=True)
-        if data_variances is not None:
```

Comment:
I was just wondering why this has been removed? Has it been added somewhere else?

Reply:
It has been abstracted into the parent `WeightedLoss`